### PR TITLE
Fixing the adtimer for different use cases.

### PR DESCRIPTION
**Pull request by paparajupavan**
1.Pausing Ad
2.For ad events which are not generated for different ad percentages

### CODE REVIEW

File: js/ooyala_ssai.js [hunk 1]
```diff
@@ -366,11 +368,21 @@ OO.Ads.manager(function()
     this.resumeAd = (ad) => {
       if (adMode) {
         _handleTrackingUrls(this.currentAd, ["resume"]);
-        if (ad && ad.ad && ad.ad.data && this.adIdDictionary[ad.ad.data.id] && isFinite(ad.duration)) {
+        if (ad && ad.ad && ad.ad.data && this.adIdDictionary[ad.ad.data.id] && _.isFinite(ad.duration)) {
+          var duration = ad.duration;
+          //Deducting the already played duration of ad  from the actual ad duration for making timer accurate
+          if(_.isFinite(this.adIdDictionary[ad.ad.data.id].startTime) && _.isFinite(this.adIdDictionary[ad.ad.data.id].pauseTime) )
```
Comment (ooyala-amohnatkin) [inline js/ooyala_ssai.js]: it's better to use destructuring to increase code readability
`const { startTime, pauseTime } = this.adIdDictionary[ad.ad.data.id];`

File: js/ooyala_ssai.js [hunk 2]
```diff
@@ -519,15 +534,22 @@ OO.Ads.manager(function()
...
-     * @method OoyalaSsai#onReplay
-     */
-    this.onReplay = () => {
-      currentOffset = 0;
-      this.currentAd = null;
-    };
+    *
+    *
+    */
+    var _getAdDuration = _.bind(function(id3Object)
```
Comment (ooyala-amohnatkin) [inline js/ooyala_ssai.js]: Please use an arrow function and remove `bind`
```
var _getAdDuration = (id3Object) => {
}
```

File: js/ooyala_ssai.js (hunk 1 above)
Comment (paparajupavan) [reply js/ooyala_ssai.js]: Incorporated the changes.

File: js/ooyala_ssai.js (hunk 2 above)
Comment (paparajupavan) [reply js/ooyala_ssai.js]: Incorporated the changes.

**Comment by paparajupavan** [comment]
> Please check that tests are passing.
> There's no `_` variable, all underscore methods are imported directly, so `_.isFinite` should be `isFinite`
> The same for `delay`

Yes.Changed the functions.

**Comment by paparajupavan** [comment]
> Please check that tests are passing.
> There's no `_` variable, all underscore methods are imported directly, so `_.isFinite` should be `isFinite`
> The same for `delay`

Modified it.

**Pull request merged**